OptimoveSDK/Sources/Classes/OptimoveConfig.swift

```diff
@@ -161,100 +189,155 @@ open class OptimoveConfigBuilder: NSObject {
         }
 
-        if let _apiKey = _apiKey,
-           let _secretKey = _secretKey,
-           let _baseUrlMap = _baseUrlMap,
-           let _region = _region
-        {
-            optimobileConfig = OptimobileConfig(
-                apiKey: _apiKey,
-                secretKey: _secretKey,
-                region: _region,
-                sessionIdleTimeout: _sessionIdleTimeout,
-                inAppConsentStrategy: _inAppConsentStrategy,
-                inAppDefaultDisplayMode: _inAppDisplayMode,
-                inAppDeepLinkHandlerBlock: _inAppDeepLinkHandlerBlock,
-                pushOpenedHandlerBlock: _pushOpenedHandlerBlock,
-                _pushReceivedInForegroundHandlerBlock: _pushReceivedInForegroundHandlerBlock,
-                deepLinkCname: _deepLinkCname,
-                deepLinkHandler: _deepLinkHandler,
-                baseUrlMap: _baseUrlMap,
-                runtimeInfo: _runtimeInfo,
-                sdkInfo: _sdkInfo,
-                isRelease: _isRelease
-            )
-        }
+        let tenantInfo: OptimoveTenantInfo? = {
+            if let _tenantToken = _tenantToken,
+               let _configName = _configName
+            {
+                return OptimoveTenantInfo(tenantToken: _tenantToken, configName: _configName)
+            }
+            return nil
+        }()
+
+        var optimobileConfig: OptimobileConfig? = {
+            if let region = region,
+               let _baseUrlMap = _baseUrlMap
+            {
+                return OptimobileConfig(
+                    credentials: credentials,
+                    region: region,
+                    sessionIdleTimeout: _sessionIdleTimeout,
+                    inAppConsentStrategy: _inAppConsentStrategy,
+                    inAppDefaultDisplayMode: _inAppDisplayMode,
+                    inAppDeepLinkHandlerBlock: _inAppDeepLinkHandlerBlock,
+                    pushOpenedHandlerBlock: _pushOpenedHandlerBlock,
+                    _pushReceivedInForegroundHandlerBlock: _pushReceivedInForegroundHandlerBlock,
+                    deepLinkCname: _deepLinkCname,
+                    deepLinkHandler: _deepLinkHandler,
+                    baseUrlMap: _baseUrlMap,
+                    runtimeInfo: _runtimeInfo,
+                    sdkInfo: _sdkInfo,
+                    isRelease: _isRelease
+                )
+            }
+            return nil
+        }()
 
         return OptimoveConfig(
             tenantInfo: tenantInfo,
             optimobileConfig: optimobileConfig
         )
     }
+}
 
-    private static func parseOptimoveCredentials(creds: String?) -> (tenantToken: String, configName: String)? {
-        guard let creds = creds,
-              let tuple = parseTuple(creds: creds),
-              let ver = tuple[0] as? Int
-        else {
-            return nil
+public extension OptimobileConfig {
+    enum Region: String, CaseIterable {
+        case EU1 = "eu-central-1"
```

Comment:
i think dont need EU1 -- it's not used for optimove tenants and would needlessly confuse people